### DISCUSSION TITLE
Is it possible to get a more user-friendly explanation of how to run this

### DISCUSSION THREAD

**Discussion by throwaway2157546** [pmqs/Fix-OneDrive-Zip#8] [Q&A]
I came here from the bitsgalore article and was hoping that this would help me since I'm trying to download my onedrive files, which are large art projects which can take multiple gigs.

Since this is a problem affecting many windows and onedrive users, I don't think the current readme.md is very descriptive for anyone who isn't already adept at what this script is.

The readme doesn't explain how to run this at all, other than that you need to install Perl, which I assume is the coding language, since the readme doesn't link to it directly. Of course, to an average user, one doesn't know which version of Perl to install (Strawberry or Activestate? Assuming you're in windows).

Then the usage section doesn't even explain where to put this command. Do you run it in command prompt/powershell? Does it have to be run from the same folder as the .zip file?

Speaking as someone who's just average tech savvy, this is an extremely intimidating and not well explained solution to a problem that sounded quite simple, when explained in the original article.

All this is to say, if this affects everyone, this solution is only accessible to maybe 3% of them who are already adept enough that they could've just edited the file themselves in a hex editor.

**Comment by pmqs** [comment]
Hey @throwaway2157546 

thanks for taking the time to provide very valuable feedback.  

Let me address your specific questions. Please say if I'm pitching this at the wrong level. I can use this as input to a windows-specific section in the README.md file.

> you need to install Perl, which I assume is the coding language, since the readme doesn't link to it directly. Of course, to an average user, one doesn't know which version of Perl to install (Strawberry or Activestate? Assuming you're in windows).

Yes, `perl` is a language. Either Strawberry or Activestate will do. You need a 64-bit build, which I see Strawberry provides. I have used Strawberry Perl  from here https://strawberryperl.com/ . Never used ActiveState but don't expect there would be an issue with using it.

> Do you run it in command prompt/powershell? 

Either `cmd` or `powershell` will do. The key point is you need a terminal window to run the script.

Once you have a terminal window available, you can check that `perl` is available by running `perl -v`. The output should look like this if you have it installed correctly

```
C:\>perl -v

This is perl 5, version 32, subversion 1 (v5.32.1) built for MSWin32-x64-multi-thread

Copyright 1987-2021, Larry Wall

Perl may be copied only under the terms of either the Artistic License or the
GNU General Public License, which may be found in the Perl 5 source kit.

Complete documentation for Perl, including FAQ lists, should be found on
this system using "man perl" or "perldoc perl".  If you have access to the
Internet, point your browser at http://www.perl.org/, the Perl Home Page.
```


> Does it have to be run from the same folder as the .zip file?

Not necessarily, but probably easiest to keep both the zipfile you need fixed and this script in the same directory. Then just run this in the terminal window

```
C:\somedir> perl fix-onedrive-zip onedrive.zip
```

Alternatively, if you want to have the script in one directory (say `C:\scripts` ) and the zip files in another (say `C:\zips`) you can run the script like this


```
C:\somedir> perl C:\scripts\fix-onedrive-zip c:\zips\onedrive.zip
```

Does that get you up & running?
